conans/model/build_info.py

```diff
@@ -41,6 +47,66 @@ def __init__(self):
         self.description = None  # Description of the conan package
         # When package is editable, filter_empty=False, so empty dirs are maintained
         self.filter_empty = True
+        self._deps = OrderedDict()
+
+    @property
+    def libs(self):
+        if self._deps:
+            deps = [v for v in self._deps.values()]
+            deps_sorted = sorted(deps, key=lambda component: len(component.deps))
+            self._libs = [dep.lib for dep in deps_sorted if dep.lib is not None]
```

Comment:
Why assigning to `self._libs` instead of returning the list (like you are doing with the `exes` or `system_deps` property)? I love things looking like the same if they are similar

Reply:
And I think that given the checks you are doing it is not safe to assign to `self._libs`

Reply:
Sure, I have changed this. Thanks